src/packet_analysis/protocol/ayiya/AYIYA.cc

```diff
@@ -0,0 +1,78 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#include "zeek/packet_analysis/protocol/ayiya/AYIYA.h"
+
+#include "zeek/packet_analysis/protocol/iptunnel/IPTunnel.h"
+
+using namespace zeek::packet_analysis::AYIYA;
+
+AYIYAAnalyzer::AYIYAAnalyzer() : zeek::packet_analysis::Analyzer("AYIYA") { }
+
+bool AYIYAAnalyzer::AnalyzePacket(size_t len, const uint8_t* data, Packet* packet)
+	{
+	if ( ! BifConst::Tunnel::enable_ayiya )
+		return false;
+
+	if ( packet->encap && packet->encap->Depth() >= BifConst::Tunnel::max_depth )
```

Comment:
Would be nice if we had some external place where this check could be done, doesn't feel like something a specific (and every) analyzer should be in charge of. That said, I can't think of a place where that would be.

Reply:
Would changing it to a method like `Packet::EncapValid()` be enough?

Reply:
Yeah, that's would help, and I still don't have a better ideas where we could centralize this, so yes.

Reply:
Yah I see what you mean, looking at it again. Even adding a function that does the check doesn't change the fact that we're still having to send the weird in all of those places. I'd move it up into `Analyzer::ForwardPacket`, but that doesn't feel like the right place to do it either.

Reply:
We decided to leave this one alone, since the original session analyzers were doing the same thing anyways. There's not really a good place to put a function that does this either.